### ISSUE TITLE
0.9.4-rc.1: Private subnet with existing nat gateway specification does not create route

### ISSUE THREAD

**Issue by drock** [coreos/kube-aws#299]
version 0.9.4-rc.1

The documentation in cluster.yaml gives this example for specifying a managed subnet to use an existing nat gateway
```
#   #
#   # Advanced: Managed private subnet with an existing NAT gateway
#   #
#   # kube-aws tries to reuse the ngw specified by id or idFromStackOutput
#   # by adding a route to the ngw to a route table managed by kube-aws
#   #
#   # Please be sure that the NGW is properly deployed. kube-aws will never modify ngw itself.
#   #
#   - name: ManagedPrivateSubnetWithExistingNGW
#     private: true
#     availabilityZone: us-west-1a
#     instanceCIDR: "10.0.1.0/24"
#     natGateway:
#       id: "ngw-xxxxxxxx"
#       # Exported output's name from another stack
#       # Only specify either id or idFromStackOutput but not both
#       #idFromStackOutput: myinfra-PrivateSubnet1
```

However, when I follow that example, my subnet is created with a route table but no route is created  is created at all.  This is my subnet specification:
```yaml
subnets:
   - name: EtcdSubnet1a
     private: true
     availabilityZone: us-east-1a
     instanceCIDR: "10.0.12.0/24"
     natGateway:
       id: "foobar"
```

The resulting CF stack template from doing `--export` has no routes for that subnet.  Searching for the string "foobar" yields no results.

In stack-template.json all nat gateway routes are added within a loop
```go
{{range $i, $ngw := .NATGateways}}
```

However, the method defined on https://github.com/coreos/kube-aws/blob/v0.9.4-rc.1/config/config.go#L1012 only adds managed nat gateways to that list.

**Comment by mumoshu** [comment]
@drock Thanks for your feedback :bow:
I've released v0.9.4-rc.2 which includes the fix for this.
Would you mind trying?

**Comment by drock** [comment]
@mumoshu Thank you for the prompt turnaround.  v0.9.4-rc.2 does appear to resolve this issue for me :smile:
